File: tags/NHI1-0.14/theLink/example/python/mulserver.py
#+
#§  \file       theLink/example/python/mulserver.py
#§  \brief      \$Id$
#§  
#§  (C) 2009 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

import sys
from pymsgque import *

class MulServer(MqS):
  def __init__(self, tmpl=None):
    self.ConfigSetServerSetup(self.ServerSetup)
    MqS.__init__(self)
  def MMUL(self):
    self.SendSTART()
    self.SendD(self.ReadD() * self.ReadD())
    self.SendRETURN()
  def ServerSetup(self):
    self.ServiceCreate("MMUL",self.MMUL)

srv = FactoryAdd("mulserver", MulServer).New();
try:
  srv.LinkCreate(sys.argv)
  srv.ProcessEvent(MqS_WAIT_FOREVER)
except:
  srv.ErrorSet()
finally:
  srv.Exit()

File: tags/NHI1-0.17/theLink/example/ruby/mulserver.rb
#+
#:  \file       theLink/example/ruby/mulserver.rb
#:  \brief      \$Id$
#:  
#:  (C) 2010 - NHI - #1 - Project - Group
#:  
#:  \version    \$Rev$
#:  \author     EMail: aotto1968 at users.berlios.de
#:  \attention  this software has GPL permissions to copy
#:              please contact AUTHORS for additional information
#:

require "rubymsgque"

class MulServer < MqS
  def initialize()
    super()
    ConfigSetServerSetup(method(:ServerSetup))
  end
  def MMUL
    SendSTART()
    SendD(ReadD() * ReadD())
    SendRETURN()
  end
  def ServerSetup()
    ServiceCreate("MMUL",method(:MMUL))
  end
end

srv = FactoryAdd("mulserver", MulServer).New()
begin
  srv.LinkCreate($0,ARGV)
  srv.ProcessEvent(MqS::WAIT_FOREVER)
rescue Exception => ex
  srv.ErrorSet(ex)
ensure
  srv.Exit()
end

